natchez-datadog/src/main/scala/com/ovoenergy/effect/natchez/Datadog.scala

```diff
@@ -113,7 +113,9 @@ object Datadog {
     } yield {
       new EntryPoint[F] {
         def root(name: String): Resource[F, Span[F]] =
-          Resource.liftF(SpanIdentifiers.create).flatMap(DatadogSpan.create(queue, names(name))).widen
+          Resource.liftF(
+            SpanIdentifiers.create.flatMap(Ref.of[F, SpanIdentifiers])
+          ).flatMap(DatadogSpan.create(queue, names(name))).widen
```

Comment:
can this not be `Resource.pure(SpanIdentifiers.create)` instead of `liftF`?

Reply:
`SpanIdentifiers.create` returns an `F[SpanIdentifiers]` as it randomly generates things if ids are missing, hence the `liftF`

Reply:
gotcha, thanks